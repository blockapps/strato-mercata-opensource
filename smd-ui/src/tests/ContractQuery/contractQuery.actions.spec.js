import {
  queryCirrusVars,
  queryCirrusVarsSuccess,
  queryCirrusVarsFailure,
  clearQueryString,
  queryCirrusSuccess,
  queryCirrusFailure,
  queryCirrus,
  addQueryFilter,
  removeQueryFilter
} from '../../components/ContractQuery/contractQuery.actions';
import { error, queryCirrusVarsMock, queryCirrusMock } from './contractQueryMock';

describe('ContractQuery: action', () => {

  describe('fetch queryCirrusVars', () => {

    test('request', () => {
      expect(queryCirrusVars('Bid')).toMatchSnapshot();
    });

    test('success', () => {
      expect(queryCirrusVarsSuccess(queryCirrusVarsMock.xabi.vars)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(queryCirrusVarsFailure(error)).toMatchSnapshot();
    });

  })

  test('clear query', () => {
    expect(clearQueryString()).toMatchSnapshot();
  });

  describe('fetch queryCirrus', () => {

    test('request', () => {
      expect(queryCirrus('Bid', undefined, "ff7ef45acb7a775018bc765b6fdeea432aaddfcd846cf6dd9442724266b1eac9")).toMatchSnapshot();
    });

    test('success', () => {
      expect(queryCirrusSuccess(queryCirrusMock)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(queryCirrusFailure(error)).toMatchSnapshot();
    });

  })

  test('add query filter', () => {
    const payload = {
      field: 'state',
      operator: 'eq',
      value: 1
    };
    expect(addQueryFilter(payload.field, payload.operator, payload.value)).toMatchSnapshot();
  });

  test('remove query filter', () => {
    expect(removeQueryFilter(1)).toMatchSnapshot();
  });

});