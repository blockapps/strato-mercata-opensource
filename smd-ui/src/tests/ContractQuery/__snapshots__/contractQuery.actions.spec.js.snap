// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractQuery: action add query filter 1`] = `
Object {
  "field": "state",
  "operator": "eq",
  "type": "ADD_QUERY_FILTER",
  "value": 1,
}
`;

exports[`ContractQuery: action clear query 1`] = `
Object {
  "type": "CLEAR_QUERY_STRING",
}
`;

exports[`ContractQuery: action fetch queryCirrus failure 1`] = `
Object {
  "error": "ERROR",
  "type": "QUERY_CIRRUS_FAILURE",
}
`;

exports[`ContractQuery: action fetch queryCirrus request 1`] = `
Object {
  "contractName": undefined,
  "queryString": "ff7ef45acb7a775018bc765b6fdeea432aaddfcd846cf6dd9442724266b1eac9",
  "tableName": "Bid",
  "type": "QUERY_CIRRUS_REQUEST",
}
`;

exports[`ContractQuery: action fetch queryCirrus success 1`] = `
Object {
  "queryResults": Array [
    Object {
      "address": "ddbfe1fe25dfbb91633c576aaadd8d3e377be47b",
      "amount": 5678,
      "id": 149,
      "name": "P_ ?%#@!:* 41769",
      "state": "1",
      "supplier": "Supplier1_1301_24669",
    },
    Object {
      "address": "5112a195a0bf46dbeac6ce2e9a2fbbed50dc3e44",
      "amount": 67,
      "id": 150,
      "name": "P_ ?%#@!:* 92348",
      "state": "2",
      "supplier": "Supplier1",
    },
    Object {
      "address": "eff065f2e1cf483daee729f1f45c92db46fe1092",
      "amount": 1067,
      "id": 151,
      "name": "P_ ?%#@!:* 44366",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "bd0a5c9f7965e3ff970b2b69eb686a9dd6d93ea3",
      "amount": 32,
      "id": 152,
      "name": "P_ ?%#@!:* 46441",
      "state": "2",
      "supplier": "Supplier0_1301_46441",
    },
    Object {
      "address": "2e126696dd113e8ef1a96f51d4fe21bdec5bb8d4",
      "amount": 32,
      "id": 153,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier1_1301_46441",
    },
    Object {
      "address": "421022d27ad192eb2979d0d102945d05b1b62186",
      "amount": 32,
      "id": 154,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier2_1301_46441",
    },
    Object {
      "address": "1bf900ac2d58688ebd9847ec9993533b4207165e",
      "amount": 5678,
      "id": 155,
      "name": "P_ ?%#@!:* 80772",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "ad601648712a607acf73cae058d4bc8601c1a724",
      "amount": 100,
      "id": 3,
      "name": "Project_ ? 1452_77996",
      "state": "2",
      "supplier": "Supplier1",
    },
  ],
  "type": "QUERY_CIRRUS_SUCCESS",
}
`;

exports[`ContractQuery: action fetch queryCirrusVars failure 1`] = `
Object {
  "error": "ERROR",
  "type": "QUERY_CIRRUS_VARS_FAILURE",
}
`;

exports[`ContractQuery: action fetch queryCirrusVars request 1`] = `
Object {
  "contractAddress": undefined,
  "contractName": "Bid",
  "type": "QUERY_CIRRUS_VARS_REQUEST",
}
`;

exports[`ContractQuery: action fetch queryCirrusVars success 1`] = `
Object {
  "type": "QUERY_CIRRUS_VARS_SUCCESS",
  "vars": Object {
    "amount": Object {
      "atBytes": 96,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "id": Object {
      "atBytes": 0,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "name": Object {
      "atBytes": 32,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
    "state": Object {
      "atBytes": 128,
      "contents": "BidState",
      "public": true,
      "type": "Label",
    },
    "supplier": Object {
      "atBytes": 64,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
  },
}
`;

exports[`ContractQuery: action remove query filter 1`] = `
Object {
  "index": 1,
  "type": "REMOVE_QUERY_FILTER",
}
`;
